src/MetadataManager/MetadataEventManager.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as vscode from 'vscode';
+
+import {Balloon} from '../Utils/Balloon';
+import {obtainWorkspaceRoot} from '../Utils/Helpers';
+import {Logger} from '../Utils/Logger';
+
+class MetadataEventQueue {
+  private inProgress: boolean = false;
```

Comment:
Is this variable is for multi-thread?
AFAIK, node.js is single thread... @.@

Reply:
As you know, node.js is single thread and the concept of event queue is for multi-thread. I agree.
But asynchronous event makes similar situation to multi-thread...
When the two event occurs and access same metadata file(.json), they get same information and overwrite file. This is reason why we add event queue.